src/search/srpso.c

```
@@ -0,0 +1,249 @@
+  if (evals) *evals = local_evals;
+
+  sr_free_matrix(pos);
+  sr_free_matrix(vel);
+  sr_free_matrix(pbest);
+  sr_free_vector(pbest_val);
+  sr_free_vector(gbest);
+
+  return 0;
+}
+
+void sr_pso(int ndim, real dpos, const char *bak_file, const char *log_file)
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'sr_pso' of similar type ('int') are easily swapped by mistake

(bugprone-easily-swappable-parameters)